lib/cmd/fh3/app/stage.js

```diff
@@ -1,60 +1,137 @@
+  exec : genericCommand(require('./runtimes'))
 };
-base.method = 'post';
-base.describe.runtime = i18n._("The Node.js runtime of your application, e.g. node06 or node08 or node010");
-base.describe.clean = i18n._("Do a full, clean stage. Cleans out all old application log files, removes cached node modules and does an 'npm install' from scratch");
-base.describe.autodeploy = i18n._("Deploy the app automatically when the app repo is updated");
-base.describe['gitRef.type'] = i18n._("Specifies if you'd like to deploy a 'branch' or 'tag'");
-base.describe['gitRef.hash'] = i18n._("The commit hash you'd like to deploy if gitRef.type is 'branch'. HEAD will use the latest code for that branch");
-base.describe['gitRef.value'] = i18n._("The name of the branch you'd like to deploy");
-base.alias.runtime = "r";
-base.alias.clean = "c";
-base.alias.autodeploy = "d";
-
-//Add custom example
-base.examples[0].cmd = 'fhc app stage --app=<appGuid> --env=<environmentName> (Defaults to branch:master)';
-
-base.preCmd = function(argv, cb) {
-  var data = {
-    env: argv.env,
-    app: argv.app,
-    gitRef: argv.gitRef || {}
-  };
-
-  // Default to master like studio
-  if (!argv.gitRef) {
+
+module.exports = {
+  'desc' : i18n._('Stages a cloud application.'),
+  'examples' :
+  [{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment>',
+    desc : i18n._("Deploy of cloud application <app-id> on <environment> (Defaults to branch:master)")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --domain=<domain>',
+    desc : i18n._("Deploy of cloud application <app-id> on <environment> and <domain>")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime>")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --clean=false',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> without Clean Stage option selected.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --autodeploy=true',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with 'Auto Deploy' option selected.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefType=<gitRef-type>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with 'branch' or 'tag' informed.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefHash=<gitRef-hash>',
+    desc : i18n._("Deploy of cloud application <cloud-app-id> on <environment> with the <runtime> with the commit hash that you'd like to deploy.")
+  },{
+    cmd : 'fhc app stage --app=<cloud-app-id> --env=<environment> --runtime=<runtime> --gitRefValue=<gitRef-value>',
```

Comment:
👍 I change for has fewer examples. But just one will be a mess because are too many options.